pkg/workflow/tasks/discover.go

```diff
@@ -140,6 +149,46 @@ func (tr *suspendTaskRunner) Pending(ctx wfContext.Context) bool {
 	return false
 }
 
+// SubTaskRunners return child step names. it could be null if the step have no sub-step
+func (tr *suspendTaskRunner) SubTaskRunners() []types.TaskRunner {
+	return nil
+}
+
+type stepGroupTaskRunner struct {
+	id             string
+	name           string
+	subTaskRunners []types.TaskRunner
+}
+
+// Name return suspend step name.
+func (tr *stepGroupTaskRunner) Name() string {
+	return tr.name
+}
+
+// SubTaskRunners return child step runners. it could be null if the step have no sub-step
+func (tr *stepGroupTaskRunner) SubTaskRunners() []types.TaskRunner {
+	return tr.subTaskRunners
+}
+
+// Run make workflow step group.
+func (tr *stepGroupTaskRunner) Run(ctx wfContext.Context, options *types.TaskRunOptions) (common.StepStatus, *types.Operation, error) {
+	phase := common.WorkflowStepPhaseRunning
+	if tr.subTaskRunners == nil {
+		phase = common.WorkflowStepPhaseSucceeded
+	}
+	return common.StepStatus{
+		ID:    tr.id,
+		Name:  tr.name,
+		Type:  types.WorkflowStepTypeStepGroup,
+		Phase: phase,
+	}, &types.Operation{}, nil
+}
+
+// Pending check task should be executed or not.
+func (tr *stepGroupTaskRunner) Pending(ctx wfContext.Context) bool {
```

Comment:
If you always return `false` in `Pending`, it means that you can not use `inputs/outputs` and `dependsOn` in stepGroup. It is also a known issue for `suspend`. Anyway, I already fixed that in #3941, and will fix that for step group as well.

Reply:
ok